devlib/module/gem5stats.py

```diff
@@ -66,6 +67,7 @@ def __init__(self, target):
         self._stats_file_path = os.path.join(target.platform.gem5_out_dir,
                                             'stats.txt')
         self.rois = {}
+        self._dump_pos_cache = {0: 0}
```

Comment:
Mmm ok. Yeah, if we can't assume the keys are inserted in incremental order, then something like a red-black tree would be the way to go. There is nothing like that in the standard lib. There is a `bintrees` package that provides an implementation. However, I'd rather hold-off on introducing an additional dependency until/unless lines 176:177 prove to be a bottle neck. I guess, we can leave them as-is for now.